src/TreeView/TreeView.test.tsx

```diff
@@ -1,50 +1,706 @@
+            <TreeView.SubTree>
+              <TreeView.Item>Child 2</TreeView.Item>
+            </TreeView.SubTree>
+          </TreeView.Item>
+          <TreeView.Item>
+            Parent 3
+            <TreeView.SubTree>
+              <TreeView.Item>Child 3</TreeView.Item>
+            </TreeView.SubTree>
+          </TreeView.Item>
+        </TreeView>
+      )
+
+      const root = getByRole('tree')
+      const parent1 = getByRole('treeitem', {name: 'Parent 1'})
+      const parent3 = getByRole('treeitem', {name: 'Parent 2'})
+
+      // Focus tree
+      root.focus()
+
+      // Press ↓ 2 times to move aria-activedescendant to parent 3
+      fireEvent.keyDown(document.activeElement || document.body, {key: 'ArrowDown'})
+
+      // aria-activedescendant should now be set to parent 3
+      expect(root).toHaveAttribute('aria-activedescendant', parent3.id)
+
+      // Press Home
+      fireEvent.keyDown(document.activeElement || document.body, {key: 'Home'})
+
+      // aria-activedescendant should now be set to parent 1
+      expect(root).toHaveAttribute('aria-activedescendant', parent1.id)
+    })
+  })
+
+  describe('End', () => {
+    it('moves aria-activedescendant to last visible item', () => {
+      const {getByRole} = render(
+        <TreeView aria-label="Test tree">
+          <TreeView.Item>
+            Parent 1
+            <TreeView.SubTree>
+              <TreeView.Item>Child 1</TreeView.Item>
+            </TreeView.SubTree>
+          </TreeView.Item>
+          <TreeView.Item>
+            Parent 2
+            <TreeView.SubTree>
+              <TreeView.Item>Child 2</TreeView.Item>
+            </TreeView.SubTree>
+          </TreeView.Item>
+          <TreeView.Item>
+            Parent 3
+            <TreeView.SubTree>
+              <TreeView.Item>Child 3</TreeView.Item>
+            </TreeView.SubTree>
+          </TreeView.Item>
+        </TreeView>
+      )
+
+      const root = getByRole('tree')
+      const parent1 = getByRole('treeitem', {name: 'Parent 1'})
+      const parent3 = getByRole('treeitem', {name: 'Parent 3'})
+
+      // Focus tree
+      root.focus()
+
+      // aria-activedescendant should be set to parent 1
+      expect(root).toHaveAttribute('aria-activedescendant', parent1.id)
+
+      // Press End
+      fireEvent.keyDown(document.activeElement || document.body, {key: 'End'})
+
+      // aria-activedescendant should now be set to parent 3
+      expect(root).toHaveAttribute('aria-activedescendant', parent3.id)
+
+      // Press → to expand parent 3
+      fireEvent.keyDown(document.activeElement || document.body, {key: 'ArrowRight'})
+
+      // Press End
+      fireEvent.keyDown(document.activeElement || document.body, {key: 'End'})
+
+      const child3 = getByRole('treeitem', {name: 'Child 3'})
+
+      // aria-activedescendant should now be set to child 3
+      expect(root).toHaveAttribute('aria-activedescendant', child3.id)
+    })
+  })
+
+  describe('Enter', () => {
```

Comment:
Should we add a test to make sure that keying "Enter" on an expandable `LinkItem` activates the link node?

Reply:
Good idea. Added ✅